src/common/watch.cpp

```diff
@@ -68,6 +68,37 @@ Watch::~Watch()
     }
 }
 
+void Watch::removeWatch()
+{
+    if ((_watchFileDescriptor() >= 0) && (_watchDescriptor >= 0))
+    {
+        if (inotify_rm_watch(_watchFileDescriptor(), _watchDescriptor) == -1) 
+        {
+            log<level::ERR>(
+                std::format("inotify_rm_watch call failed with ErrNo[{}] "
+                            "ErrMsg[{}]",
+                            errno, strerror(errno))
+                    .c_str());
+            throw std::runtime_error("Failed to remove watch");
+        }
+    }
+}
+
+void Watch::addWatch()
+{
```

Comment:
we can call this method in constrcutor also, what happens to sd_event_add_io, shouldn;t this also be changed as it will be polling on old file descriptor of the watch.

Reply:
calling the same method in constructor.  It does not change the watchDescriptor, it only adds/removes the files to be watched.